codec/packer.go

```diff
@@ -94,65 +70,65 @@ func (p *Packer) UnpackFixedBytes(size int, dest *[]byte) {
 // UnpackBytes adds an err ErrFieldNotPopulated to the Packer.
 func (p *Packer) UnpackBytes(limit int, required bool, dest *[]byte) {
 	if limit >= 0 {
-		*dest = p.p.UnpackLimitedBytes(uint32(limit))
+		*dest = p.Packer.UnpackLimitedBytes(uint32(limit))
 	} else {
-		*dest = p.p.UnpackBytes()
+		*dest = p.Packer.UnpackBytes()
 	}
 	if required && len(*dest) == 0 {
 		p.addErr(fmt.Errorf("%w: Bytes field is not populated", ErrFieldNotPopulated))
 	}
 }
 
 func (p *Packer) PackUint64(v uint64) {
-	p.p.PackLong(v)
+	p.Packer.PackLong(v)
 }
 
 func (p *Packer) UnpackUint64(required bool) uint64 {
-	v := p.p.UnpackLong()
+	v := p.Packer.UnpackLong()
 	if required && v == 0 {
 		p.addErr(fmt.Errorf("%w: Uint64 field is not populated", ErrFieldNotPopulated))
 	}
 	return v
 }
 
 func (p *Packer) PackInt64(v int64) {
-	p.p.PackLong(uint64(v))
+	p.Packer.PackLong(uint64(v))
 }
 
 func (p *Packer) UnpackInt64(required bool) int64 {
-	v := p.p.UnpackLong()
+	v := p.Packer.UnpackLong()
 	if required && v == 0 {
 		p.addErr(fmt.Errorf("%w: Int64 field is not populated", ErrFieldNotPopulated))
 	}
 	return int64(v)
 }
 
-func (p *Packer) PackInt(v int) {
-	p.p.PackInt(uint32(v))
+func (p *Packer) PackInt(v int32) {
+	p.Packer.PackInt(uint32(v))
 }
 
-func (p *Packer) UnpackInt(required bool) int {
-	v := p.p.UnpackInt()
+func (p *Packer) UnpackInt(required bool) int32 {
+	v := p.Packer.UnpackInt()
 	if required && v == 0 {
 		p.addErr(fmt.Errorf("%w: Int field is not populated", ErrFieldNotPopulated))
 	}
-	return int(v)
+	return int32(v)
```

Comment:
It's not clear to me why this ever did this type conversion. I think we should just use the `uint32` type that the underlying packer is giving us especially because everywhere we're using it, we could use `uint32`